src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -33,7 +33,11 @@ class SchemaService(private val options: Neo4jOptions, private val driverCache:
 
   private def structForNode(labels: Seq[String] = options.nodeMetadata.labels): StructType = {
     var structFields: mutable.Buffer[StructField] = (try {
-        val query = "CALL apoc.meta.nodeTypeProperties({ includeLabels: $labels })"
+        val query =
+          """CALL apoc.meta.nodeTypeProperties({ includeLabels: $labels })
+            |YIELD propertyName, propertyTypes
+            |RETURN DISTINCT propertyName, propertyTypes
```

Comment:
I think this won't work, because you'll miss type information.

Imagine this scenario:
* `(:Person { prop: 25 })`
* `(:Person:Player { prop: "hello" })`
* `(:Person:Player:Weirdo { prop: true })`

If you get the distinct propertyName, won't you get only one of those property type lines, basically at random?   If you want to take a distinct propertyName, you need to collect/merge all of the various types they can have.   In the good case, if you collect all of the various types you'll end up with [String, String, String] and then it's easy.   But you could still end up with [String, Long, Boolean]

Reply:
If the connector gets wrong metadata it's going to be quite hard to debug, since APOC is only one of the techniques.  Then we will probably start to think that it's a problem with APOC, when it's about local user database state

Reply:
That case we managed it case APOC is not present, we will coerce all to Strings, I think that we need to have the same strategy here, wdyt?

Reply:
If the types differ, "coerce everything to string" is right.  But the types might be the same across the different labels.  So if the total type collection is [Long, Long, Long, Long] -- then of course use long.  ;)

Reply:
sure I meant the use-case: `[String, Long, Boolean]` :D